contracts/MechFixedPriceBase.sol

```diff
@@ -18,13 +18,21 @@ abstract contract MechFixedPriceBase is OlasMech {
     /// @dev Performs actions before the delivery of a request.
     /// @param data Self-descriptive opaque data-blob.
     /// @return requestData Data for the request processing.
-    function _preDeliver(address, uint256, bytes memory data) internal virtual override returns (bytes memory requestData) {
```

Comment:
Requester addresses are out of the equation for mechs themselves - things are much clearer now.